src/copy-paste.js

```diff
@@ -0,0 +1,347 @@
+define([
+    'jquery',
+    'underscore',
+    'vellum/mugs',
+    'vellum/tsv',
+    'vellum/core'
+], function (
+    $,
+    _,
+    mugs,
+    tsv
+) {
+    var PREAMBLE = ["Form Builder clip", "version 1"],
+        vellum,
+        offScreen = {top: -10000, left: -10000},
+        hiddenTextarea = $('<textarea></textarea>').css({
+            position: 'absolute',
+            width: 0,
+            height: 0
+        }).css(offScreen).appendTo('body');
+
+    function focusTextarea($focus, value) {
+        if ($focus.length === 0) {
+            $focus = $("body");
+        }
+        hiddenTextarea.css({top: $focus.offset().top});
+        hiddenTextarea.val(value);
+        hiddenTextarea.focus();
+        hiddenTextarea.select();
+    }
+
+    function unfocusTextarea($focus, clear) {
+        $focus.focus();
+        var value = hiddenTextarea.val();
+        if (clear && value) {
+            // HACK fix intermittent multiple-paste on Chrome (timing related?)
+            hiddenTextarea.val("");
+        }
+        return value;
+    }
+
+    function onCut(opts) {
+        var $focus = $(':focus');
+        if ($focus.is('.jstree-anchor')) {
+            var text = opts.cut();
+            if (text) {
+                focusTextarea($focus, text);
+                setTimeout(function () {
+                    unfocusTextarea($focus);
+                }, 10);
```

Comment:
Yeah in the commit @millerdev linked to, I'd spent a lot of time coming up with code that works on both firefox and chrome, because they have different exact rules about when copy/paste events are fired that I don't currently recall. My guess is that if you replaced the timeout expression with just `unfocusTextarea($focus);`, it would break on one of chrome or firefox. (Like timeout of 10ms is to wait for the default action to happen and then apply this on top of it, or something.)